src/ClientContext.ts

```diff
@@ -539,4 +466,18 @@ export class ClientContext {
 
     return false;
   }
+
+  private async setHeaders(requestContext: RequestContext) {
+    requestContext.headers = await getHeaders({
```

Comment:
same deal here as the setEndpointCall. I'd prefer something functional and this is an unnecessary perf hit with the extra await call

Reply:
changed this to buildHeaders and didn't make it mutate in here. I need to clean up the getHeaders code to not take such a weird bag of properties :/